### ISSUE TITLE
`if_then_some_else_none` invalid suggestion

### ISSUE THREAD

**Issue by allan2** [rust-lang/rust-clippy#13407]
### Summary

`if_then_some_else_none`  is giving an invalid suggestion for the case below.

```rs
fn opt_bool_s(s: &str) -> Option<bool> {
    if s == "1" {
        Some(true)
    } else {
        None
    }
}
```

```
warning: this could be simplified with `bool::then`
   --> foo/src/lib.rs:629:2
    |
629 | /     if s == "1" {
630 | |         Some(true)
631 | |     } else {
632 | |         None
633 | |     }
    | |_____^ help: try: `(s == "1").then(true)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_then_some_else_none
```

The suggestion is invalid code.
```
expected a `FnOnce()` closure, found `bool`
the trait `FnOnce()` is not implemented for `bool`
wrap the `bool` in a closure with no arguments: `|| { /* code */ }`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fhome%2Fal%2Fdesktop%2Faigu%2Fcrates%2Fproxy%2Fsrc%2Flib.rs)
lib.rs(629, 16): required by a bound introduced by this call
bool.rs(59, 23): required by a bound in `core::bool::<impl bool>::then`
```

## Expected behaviour

While `(s == "1").then(|| true)` works, the suggestion should be `(s == "1").then_some(true)`.

See the emitted warning for the following function:
```rs
fn opt_bool(cond: bool) -> Option<bool> {
    if cond {
        Some(true)
    } else {
        None
    }
}
```

```
warning: this could be simplified with `bool::then_some`
   --> crates/proxy/src/lib.rs:641:5
    |
641 | /     if cond {
642 | |         Some(true)
643 | |     } else {
644 | |         None
645 | |     }
    | |_____^ help: try: `cond.then_some(true)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#if_then_some_else_none
```

### Version

```text
rustc 1.83.0-nightly (c52c23b6f 2024-09-16)
binary: rustc
commit-hash: c52c23b6f44cd19718721a5e3b2eeb169e9c96ff
commit-date: 2024-09-16
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
```
